### PR TITLE
Moved auth out of extensions subpackage

### PR DESCRIPTION
**Pull request by andamian**
`auth` follows IVOA standards, hence it should reside in its own sub-package.

I wasn't sure whether it should be in `pyvo.auth` or `pyvo.dal.auth`. It's in the former for now.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -19,6 +19,8 @@
 
 - Fix waveband option in registry.regsearch [#175]
 
+- Move auth subpackage to permanent location [#178]
```
Comment (bsipocz) [inline CHANGES.rst]: bike shedding (but that is all I can offer here): I don't think this needs a changelog entry as it changes something that has never been released to the wild, so the users wouldn't know the location was something else originally.

Comment (cbanek) [reply CHANGES.rst]: Yeah I'm with you on this one, I'd take it out, and just mark this pr as no-changelist-required

**Comment by cbanek** [comment]
Okay let's get rid of the changelog entry, other than that this looks good.  Thanks for handling this.

Although I will have to say, I _really_ don't like taking things like this last moment.  In the future, I'd really prefer to never be under the gun on PRs for a release.

File: CHANGES.rst (hunk 1 above)
Comment (andamian) [reply CHANGES.rst]: Good point. Reverted.

**Pull request merged**